packages/legacy-credentials/src/utils.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) 2018-2023, BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+import { CType } from '@kiltprotocol/core'
+import { PartialClaim } from '@kiltprotocol/types'
+import { SDKErrors } from '@kiltprotocol/utils'
+
+/**
+ * Produces JSON-LD readable representations of [[IClaim]]['contents']. This is done by implicitly or explicitly transforming property keys to globally unique predicates.
+ * Where possible these predicates are taken directly from the Verifiable Credentials vocabulary. Properties that are unique to a [[CType]] are transformed into predicates by prepending the [[CType]][schema][$id].
+ *
+ * @param claim A (partial) [[IClaim]] from to build a JSON-LD representation from. The `cTypeHash` property is required.
+ * @param expanded Return an expanded instead of a compacted representation. While property transformation is done explicitly in the expanded format, it is otherwise done implicitly via adding JSON-LD's reserved `@context` properties while leaving [[IClaim]][contents] property keys untouched.
+ * @returns An object which can be serialized into valid JSON-LD representing an [[IClaim]]'s ['contents'].
+ */
+export function jsonLDcontents(
+  claim: PartialClaim,
+  expanded = true
+): Record<string, unknown> {
+  const { cTypeHash, contents, owner } = claim
+  if (!cTypeHash) throw new SDKErrors.CTypeHashMissingError()
+  const vocabulary = `${CType.hashToId(cTypeHash)}#`
+  const result: Record<string, unknown> = {}
+  if (owner) result['@id'] = owner
+  if (!expanded) {
+    return {
+      ...result,
+      '@context': { '@vocab': vocabulary },
```

Comment:
I would put the `@context` last, as it is possible that the `contents` contains a `@context` property. At the very least we should make sure that's not a key of the `contents` object, otherwise the context would get overridden.

Reply:
The same applies for `result`, really. Maybe `contents` can be put first, then `result` and then our `@context`.

Reply:
https://github.com/KILTprotocol/sdk-js/pull/786/commits/7cb673598639334c520a65606b7973e4dac4caf4